### PR TITLE
good auction gen

### LINKED ISSUE #39: Generator for `Good AuctionTest`
These should be reasonably straightforward and arbitrary-like, maybe some caveats around invariants

### PR DESCRIPTION
**Pull request by Geometer1729**
closes #39 
Also pointing to `brian/type-holes` cause I still need it for hls.
The `good data adjuncts` test passes on `brian/auction-testable` where I've merged this and the script context adjunction branch

### CODE REVIEW

**Comment by AriFordsham** [comment]
Could you integrate the state initialization from #49? The next step would be to get actual scripts tests working, after #46 is complete.

**Comment by AriFordsham** [comment]
I'm going to work on the `GoodGen` typeclass; I guess this should wait for that?

**Comment by Geometer1729** [comment]
> I'm going to work on the `GoodGen` typeclass; I guess this should wait for that?

Makes sense. Feel free to leave a bunch of generic instances I can plumb with the functions here.
ie. `GenGood Shouldn'tExist a` `GenGood a => GenGood (EitherOr bad good)`

File: examples/AuctionGen.hs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  AuctionTestRedeemer (TestRedeemerBid),
+  Bid (Bid),
+  SelfOutput (SelfOutput),
+ )
+
+import Hedgehog.Plutus.TestData (Good, TestData (generalise))
+import Hedgehog.Range qualified as Range
+
+genGoodAuctionTest :: GenContext (Good AuctionTest)
+genGoodAuctionTest =
```
Comment (AriFordsham) [inline examples/AuctionGen.hs:25]: The idea was to construct a `Generics.SOP` type directly. That way, the data is guaranteed correct by construction. I agree this is very lacking ergonomically; I will have to come up with a solution.

**Pull request closed without merging**